### ISSUE TITLE
[Bug] ttnn.asinh_bw returns a zero gradient for |x| > 2^64, 24% of the bfloat16 domain: 1 + x*x overflows before the rsqrt

### ISSUE THREAD

**Issue by badnikhil** [tenstorrent/tt-metal#55725]
### Describe the bug

`ttnn.asinh_bw` returns a gradient of exactly `0` for every `|x| > 2^64`, which is **15,874 of the 65,536 bfloat16 bit patterns — 24% of the domain** — although `d/dx asinh(x) = 1/sqrt(1 + x^2)` tends to `1/|x|` and stays a normal float for every representable input up to `2^126`.

The composite squares the input before the reciprocal square root
([`unary_backward.cpp#L1043-L1058`](https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp#L1043-L1058)):

```cpp
std::vector<EltwiseUnaryWithParam> ops_chain = {
    EltwiseUnaryWithParam{UnaryOpType::SQUARE},
    EltwiseUnaryWithParam{UnaryOpType::ADD_UNARY_SFPU, 1.0f},
    EltwiseUnaryWithParam{UnaryOpType::RSQRT}};
Tensor grad_result =
    ttnn::multiply(grad, ttnn::unary_chain(input, ops_chain, output_mem_config), std::nullopt, output_mem_config);
```

`x * x` overflows to `+inf` for `|x| > sqrt(FLT_MAX) = 1.8446743e19`, `inf + 1` is `inf`, and `rsqrt(inf)` is `0`. The true derivative at the smallest overflowing input, `x = 2^64`, is `5.42e-20` — nowhere near the bottom of the float range — and it is still `1.18e-38`, a normal float, at `x = 2^126`.

The same overflow is in `acosh_bw` (`rsqrt(x*x - 1)`), and in a much narrower form in `atan_bw` / `atanh_bw` (`1/(1 +/- x^2)`, where the true result is already subnormal past `2^63`, so only 2 bit patterns are affected). `acosh_bw` is being rewritten by #53885, so this report does not propose touching it.

### Repro

```python
import torch, ttnn

device = ttnn.open_device(device_id=0)
x = torch.tensor([[1.0, 1e10, 1.8446744e19, 3.6893488e19, 1e25, 1e30, 8.507059e37]], dtype=torch.float32)
t = ttnn.from_torch(x, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
g = ttnn.from_torch(torch.ones_like(x), dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)

print(ttnn.to_torch(ttnn.asinh_bw(g, t)[0]))
# tensor([[7.0711e-01, 1.0000e-10, 0., 0., 0., 0., 0.]])
xd = x.double()
print(1.0 / torch.sqrt(1.0 + xd * xd))
# tensor([[7.0711e-01, 1.0000e-10, 5.4210e-20, 2.7105e-20, 1.0000e-25, 1.0000e-30, 1.1755e-38]])
```

### Observed vs expected

float32, gradient `1`, golden `1/sqrt(1 + x^2)` in float64. ULP is the integer distance between float32 bit patterns:

| input | got | want | ULP |
|---|---|---|---|
| 1 | 0.7071068 | 0.7071068 | 0 |
| 1e10 | 9.999999e-11 | 1e-10 | 1 |
| 1.8446744e19 (2^64) | **0** | 5.421011e-20 | 528,482,304 |
| 3.6893488e19 (2^65) | **0** | 2.710505e-20 | 520,093,696 |
| 1e25 | **0** | 1e-25 | 368,547,464 |
| 1e30 | **0** | 1e-30 | 228,737,632 |
| 8.507059e37 (2^126) | **0** | 1.175494e-38 | 8,388,608 |

bfloat16, gradient `1`, ULP between bfloat16 bit patterns:

| input | got | want | ULP |
|---|---|---|---|
| 1.8446744e19 (2^64) | **0** | 5.421011e-20 | 8,064 |
| 3.6893488e19 (2^65) | **0** | 2.710505e-20 | 7,936 |
| 9.97364e24 | **0** | 1.001664e-25 | 5,624 |
| 1.00026e30 | **0** | 9.984021e-31 | 3,490 |
| 8.507059e37 (2^126) | **0** | 1.175494e-38 | 128 |

The relative error over the whole band is 100%.

### Affected input range

All 65,536 bfloat16 bit patterns as the input, gradient `1`, counting inputs whose exact gradient is a normal float but whose result is exactly zero (subnormal inputs excluded):

| dtype | zeros where the golden is normal | of the domain | affected range | max ULP | mean ULP |
|---|---|---|---|---|---|
| bfloat16 | 15,874 | 24.2% | `2^64 <= \|x\| <= 8.51e37` | 8,064 | 996.38 |
| float32 | 15,874 | 24.2% | `2^64 <= \|x\| <= 8.51e37` | 528,482,304 | 65,295,046 |

### Which architectures

All of them. `asinh_bw` is a host-side composite in `ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp`, so Wormhole and Blackhole build the same op graph and produce the same values.

### How it was verified

ttsim Blackhole v1.10.6 (`TT_METAL_SIMULATOR=libttsim_bh.so`, slow dispatch). Exhaustive sweep of all 65,536 bfloat16 bit patterns as the input with an all-ones gradient, in both bfloat16 and float32, against `1/sqrt(1 + x^2)` evaluated in float64. Subnormal inputs (flushed by the hardware) are excluded.

`test_backward_asinh.py::test_bw_asinh` does not catch it: it draws inputs from `[-100, 100]`, twenty orders of magnitude below the overflow.

![asinh_bw before/after](https://raw.githubusercontent.com/badnikhil/tt-metal/pr-assets/tt-metal-asinh-bw.png)

**Comment by borjamoskv** [comment]
/assign

### Asymptotic Rescaling to Prevent $x^2$ Overflow for #55725

I would like to claim and resolve this issue. We have implemented and verified the asymptotic safeguard for `ttnn.asinh_bw` in `ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.cpp`.

**Mathematical Analysis & Root Cause:**
Currently, `asinh_bw` evaluates $1/\sqrt{1 + x^2}$ by chaining `SQUARE -> ADD 1.0 -> RSQRT`. For $|x| > 2^{64}$ (which constitutes 24.2% of the bfloat16 domain), $x^2$ overflows to `+inf`, causing `inf + 1 = inf` and `rsqrt(inf) = 0.0`. However, the true derivative $\frac{d}{dx} \text{asinh}(x) = 1/\sqrt{1 + x^2} \sim 1/|x|$ remains a normal float up to $|x| = 2^{126}$.

**Solution & Quantifiable Impact:**
For $|x| \ge 2^{60}$, $1/\sqrt{1 + x^2}$ is bit-identical to $1/|x|$ in float32:
- Utilizes `ttnn::where` with an asymptotic fallback to `ttnn::reciprocal(abs(x))` when magnitude exceeds overflow thresholds, leaving the fused `unary_chain` intact for $|x| < 2^{60}$.
- Completely recovers the gradient for 15,874 bfloat16 / float32 bit patterns across $2^{64} \le |x| \le 8.51 \times 10^{37}$.

The patch and regression tests are verified and committed on `borjamoskv/tt-metal:fix/asinh-bw-asymptotic-rescale` and ready for instant PR upon assignment.
